lib/rules/no-wait-for-side-effects.ts

```diff
@@ -204,10 +214,58 @@ export default createTestingLibraryRule<Options, MessageIds>({
 				return;
 			}
 
-			getSideEffectNodes(node.body).forEach((sideEffectNode) =>
+			const sideEffects = getSideEffectNodes(node.body);
+			sideEffects.forEach((sideEffectNode) =>
 				context.report({
 					node: sideEffectNode,
 					messageId: 'noSideEffectsWaitFor',
+					fix(fixer) {
+						const { parent: callExpressionNode } = node.parent;
+						const targetNode = isAwaitExpression(callExpressionNode.parent)
+							? callExpressionNode.parent
+							: callExpressionNode;
+
+						const lines = sourceCode.getText().split('\n');
+						const line = lines[targetNode.loc.start.line - 1];
+						const indent = line.match(/^\s*/)?.[0] ?? '';
+						const sideEffectLines = lines.slice(
+							sideEffectNode.loc.start.line - 1,
+							sideEffectNode.loc.end.line
+						);
+						const sideEffectNodeText = sideEffectLines.join('\n').trimStart();
+						if (
+							sideEffects.length === node.body.length &&
+							sideEffects.length === 1
+						) {
+							const tokenAfter = sourceCode.getTokenAfter(targetNode);
+							return [
+								fixer.insertTextBefore(targetNode, sideEffectNodeText),
+								tokenAfter?.value === ';'
```

Comment:
thought(non-blocking): I'd say this is a brittle implementation, since the `;` operator is optional. However, from what I've seen in the tests, the basic scenarios are covered, so I'm happy to go ahead with this and iterate later if necessary.